### ISSUE TITLE
Button interrupts

### ISSUE THREAD

**Issue by charlespax** [PaxInstruments/t400-firmware#109]
Some of the buttons may need to be connected to interrupt pins to ensure the device can operate in logging and not logging states. We should do a review to confirm buttons are connected to interrupt pins where necessary.

My understanding is that there is an interrupt loop that polls all the buttons at a set interval. I wonder if battery life would be improved if the behavior changed during logging mode. Maybe we can make it so the device goes into sleep and only checks the hardware interrupts.

In Rev 0.9 PCB, button connection assignment as below:
SW_A (Logging start/stop) ---- PIN21 (TXD/INT3) PD3 interrupt INT3
SW_B (Logging deltaT) ---- PIN20 (RXD/INT2) PD2 interrupt INT2
SW_C (C,F,K) ---- PIN32 (ICP3/CLK0/OC4A) PC7
SW_D (Toggle channel) ---- PIN40 (ADC1) PF1
SW_E (Backlight ON/OFF) ---- PIN41 (ADC0) PF0

Review in t400-electronics https://github.com/PaxInstruments/t400-electronics/issues/114

For previous discussion see https://github.com/PaxInstruments/t400-electronics/issues/60

**Comment by charlespax** [comment]
Okay, we are putting all the buttons and RTC_INT on interrupts. https://github.com/PaxInstruments/t400-electronics/issues/143.

**Comment by charlespax** [comment]
Since all the buttons will be on interrupts and Start/Stop starts and stops logging (https://github.com/PaxInstruments/t400-firmware/issues/45), we can probably close this.

I've opened an issue for implementing the interrupt functionality on the Start/Stop button (https://github.com/PaxInstruments/t400-firmware/issues/144). We should have a seperate issue(s) if we want more than one button to interrupt logging start/stop. Closing issue.
